src/Mod/Sketcher/Gui/EditModeConstraintCoinManager.cpp

```diff
@@ -1919,8 +1919,10 @@ void EditModeConstraintCoinManager::drawConstraintIcons(const GeoListFacade & ge
 
         if (dynamic_cast<SoZoomTranslation *>(translationPtr))
             absPos = static_cast<SoZoomTranslation *>(translationPtr)->abPos.getValue();
-        else
+        else if (translationPtr)

```

Comment:
It would be great to track who is calling drawConstraintIcons when this pointer is null.

The reason why this pointer is null should be related to the type of the constraints having changed. If this is the case, a better fix is to just stop the execution there, if the type of the constraints is different (same fix as last time).

If for any reason, that is not possible and no other solution is obvious, I would move this if block four lines up so that it also catches a null pointer when using the SoZoomTranslation (symmetry constraints).

In any case, thank you very much for tracking this one down. :)

Reply:
For the second you are absolutely right about dynamic_cast.

For the first, the thing is that if the constraint type does not match, we should break out the loop and stop processing constraint drawing. If that solves the issue, is a much better solution than checking all the translation pointers over the whole function.

So maybe I should have written: Could you check if checking the type of the constraint (as in the other commit) fixes the issue? If yes, could you please bring this solution in? If no, just let me know.

Sorry for the back and forth and thanks for taking care of it.

Reply:
> So maybe I should have written: Could you check if checking the type of the constraint (as in the other commit) fixes the issue? If yes, could you please bring this solution in? If no, just let me know.

Got it. It does seem to fix the issue so used that instead and rebased.

> Sorry for the back and forth and thanks for taking care of it.

No problem. Thanks for reviewing.